blocks/extensions/__init__.py

```diff
@@ -128,6 +128,24 @@ def on_interrupt(self):
         pass
 
 
+class CallbackName(str):
```

Comment:
I see your point, but yeah, we need to be consistent then and rename the arguments to callback_name. Currently it's confusing that the callback argument is a CallbackName instance (although I'm also fine with just callback, because it's shorter and the distinction isn't very important).